community/front-end/ofe/website/ghpcfe/views/clusters.py

```diff
@@ -470,6 +470,27 @@ def form_valid(self, form):
                 try:
                     for part in parts:
                         part.vCPU_per_node = machine_info[part.machine_type]["vCPU"] // (1 if part.enable_hyperthreads else 2)
+
+                        cpu_count_str = part.machine_type.split('-')[-1]
+                        lssd_cpu_count_str = part.machine_type.split('-')[-2]
+                        if cpu_count_str.isdigit():
+                            cpu_count = int(cpu_count_str)
+                            logger.info(f"CPU count: {cpu_count}, VM type: {part.machine_type}.")
+                        elif lssd_cpu_count_str.isdigit():
+                            cpu_count = int(lssd_cpu_count_str)
+                            logger.info(f"CPU count: {cpu_count}, VM type: {part.machine_type}.")
+                        else:
+                            raise ValidationError(f"Cannot extract vCPU count from machine type {part.machine_type}")
```

Comment:
I fear this might be overly restrictive. What about `e2-medium` or `a2-ultragpu-2g`. Is it possible to make this a warning instead of an error? Maybe only perform validation when extract is successful. 

PS, I believe the vcpu count is accessible  via API. aka `gcloud compute machine-types list`. I am not suggesting to use this gcloud command, just that I suspect there is some equivalent in the python client.

Reply:
Updated the logic, it was slightly over engineered. vcpu count extracting from API is on our tasks list, we are working on better way of dynamically pulling such information.